components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -648,6 +652,95 @@ public List<Claim> getMappedExternalClaimsForLocalClaim(String localClaimURI, St
+                ClaimConstants.REQUIRED_PROPERTY,
+                ClaimConstants.READ_ONLY_PROPERTY
+        };
+        for (String propertyKey: allowedProfilePropertyKeys) {
+            syncAttributeProfileProperties(claimProperties, allowedClaimProfiles, propertyKey);
+        }
+    }
+
+    /**
+     * Remove the profile properties if the profile property value is the same as the global property value.
+     * If all the profile properties have the same value, change the global property value to the profile value.
+     *
+     * @param claimProperties      Map of claim properties to be updated.
+     * @param allowedClaimProfiles Set of allowed claim profiles.
+     * @param propertyKey          Property key to be updated.
+     */
+    private void syncAttributeProfileProperties(Map<String, String> claimProperties, Set<String> allowedClaimProfiles,
+                                                String propertyKey) {
+
+        String globalValue = claimProperties.get(propertyKey);
+        boolean allProfileValuesMatch = true;
+        String consistentProfileValue = null;
+
+        for (String profileName : allowedClaimProfiles) {
+            String profilePropertyKey = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profileName +
+                    ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
+            String profilePropertyValue = claimProperties.get(profilePropertyKey);
+
+            // Remove the profile property if it is the same as the global property.
+            if (StringUtils.equals(globalValue, profilePropertyValue)) {
+                claimProperties.remove(profilePropertyKey);
+                allProfileValuesMatch = false;
+                continue;
+            }
+
+            // If a mismatch is already detected, no need to check further for consistency.
+            if (!allProfileValuesMatch) {
+                continue;
+            }
+
+            if (StringUtils.isBlank(profilePropertyValue)) {
+                allProfileValuesMatch = false;
+                continue;
+            }
+            if (consistentProfileValue == null) {
+                consistentProfileValue = profilePropertyValue;
+            } else if (!StringUtils.equals(consistentProfileValue, profilePropertyValue)) {
+                allProfileValuesMatch = false;
+            }
+        }
+
+        // If all the profiles have the same value change the global property value to the profile value.
+        if (allProfileValuesMatch && StringUtils.isNotBlank(consistentProfileValue)) {
+            claimProperties.put(propertyKey, consistentProfileValue);
+
+            // Remove the profile properties.
+            allowedClaimProfiles.forEach(profile -> {
+                String key = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profile +
+                        ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
+                claimProperties.remove(key);
+            });
```

Comment:
Isn't this operation is already executed by Line 710.

Reply:
No In line 710 what we do is, if profile specific value is similar to global value, we remove the profile specific value. Here what we do is, if all the profile values are equal then change the global value in line 733, then remove all the profile specific values, since global value is changed accordingly.